### PR TITLE
Update NavigationObstacle doc for avoidance rework

### PR DESCRIPTION
**Pull request by smix8**
Updates NavigationObstacle doc for changes in avoidance rework pr https://github.com/godotengine/godot/pull/69988.

*Docs team note: this is 4.x material, so will mean we need to start to separate `4.0` and `master`  
See also: https://github.com/godotengine/godot-docs/pull/7350*

### CODE REVIEW

File: tutorials/navigation/navigation_using_navigationobstacles.rst [hunk 1]
```diff
@@ -3,52 +3,92 @@
...
+    NavigationObstacles do not change or influence the pathfinding in any way.
+    NavigationObstacles only affect the avoidance velocities of agents controlled by avoidance.
 
-To use NavigationObstacles for avoidance, place a NavigationObstacle2D/3D node
-below a Node2D/3D inheriting parent node. While the obstacle node has an
-option to ``estimate_radius`` from child collisions, prefer to set a
-more reliable manual ``radius`` value. If estimated, the obstacle will use
-a radius that encapsulates the entire parent node which can result in a very large
-radius value if the parent is not a circle shape but e.g. a long rectangle shape.
+NavigationObstacles can be used in two distinct ways, either as static obstacles or as dynamic obstacles.
```
Comment (paddy-exe) [inline tutorials/navigation/navigation_using_navigationobstacles.rst:16]: Could be probably shortened to just "either as static or dynamic obstacles"

**Review by paddy-exe — COMMENTED** [review]
Left some comments with style suggestions but looks very good overall

**Comment by smix8** [comment]
I slightly changed the entry while adding paddy's suggestions.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thanks for creating and providing docs for your work! Merged. :)

**Pull request merged**